src/blockchain/esplora/ureq.rs

```diff
@@ -127,35 +127,40 @@ impl WalletSync for EsploraBlockchain {
                         .take(self.concurrency as usize)
                         .cloned();
 
-                    let handles = scripts.map(move |script| {
-                        let client = self.url_client.clone();
-                        // make each request in its own thread.
-                        std::thread::spawn(move || {
-                            let mut related_txs: Vec<Tx> = client._scripthash_txs(&script, None)?;
-
-                            let n_confirmed =
-                                related_txs.iter().filter(|tx| tx.status.confirmed).count();
-                            // esplora pages on 25 confirmed transactions. If there's 25 or more we
-                            // keep requesting to see if there's more.
-                            if n_confirmed >= 25 {
-                                loop {
-                                    let new_related_txs: Vec<Tx> = client._scripthash_txs(
-                                        &script,
-                                        Some(related_txs.last().unwrap().txid),
-                                    )?;
-                                    let n = new_related_txs.len();
-                                    related_txs.extend(new_related_txs);
-                                    // we've reached the end
-                                    if n < 25 {
-                                        break;
+                    #[allow(clippy::needless_collect)]
```

Comment:
:bulb: To avoid this you could write a for loop and manually push the threads into a vec. It might not be idiomatic to spawn threads in an iterator. Not a blocker for me. It's probably worth a short comment as to why the ignore here is a good idea otherwise.

Reply:
good idea, much simpler